src/systems/optical_tactile_plugin/Visualization.hh

```diff
@@ -0,0 +1,125 @@
+#define IGNITION_GAZEBO_SYSTEMS_OPTICAL_TACTILE_PLUGIN_VISUALIZATION_HH_
+
+#include <memory>
+#include <string>
+
+#include <ignition/gazebo/config.hh>
+#include <ignition/gazebo/Export.hh>
+#include <ignition/gazebo/System.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE
+{
+namespace systems
+{
+namespace optical_tactile_sensor
+{
+  // This class handles the configuration and process of visualizing the
+  // different elements needed for the Optical Tactile Sensor plugin
+  class OpticalTactilePluginVisualization
+  {
+    /// \brief Constructor
+    /// \param[in] _contactRadius Value of the contactRadius attribute
+    /// \param[in] _forceRadius Value of the forceRadius attribute
+    /// \param[in] _forceLength Value of the forceLength attribute
+    /// \param[in] _cameraUpdateRate Value of the cameraUpdateRate attribute
+    /// \param[in] _depthCameraOffset Value of the depthCameraOffset attribute
+    /// \param[in] _visualizationResolution Value of the
+    /// visualizationResolution attribute
+    public: OpticalTactilePluginVisualization(double &_contactRadius,
+      double &_forceRadius, double &_forceLength, float &_cameraUpdateRate,
+      ignition::math::Pose3f &_depthCameraOffset,
+      int &_visualizationResolution);
+
+    /// \brief Destructor
+    public: ~OpticalTactilePluginVisualization();
+
+    /// \brief Initialize the messages needed to request the /marker service
+    /// \param[in] _modelName Name of the model
+    /// \param[in] _sensorSize Size of the contact sensor
+    public: void CreateMarkersMsgs(std::string &_modelName,
+      ignition::math::Vector3d &_sensorSize);
+
+    /// \brief Visualize the collision geometry of the sensor. This can be
+    /// helpful when debbuging, given that there shouldn't be a visual tag
+    /// in the plugin's model
+    /// \param[in] _sensorPose Pose of the optical tactile sensor
+    public: void VisualizeSensor(ignition::math::Pose3f &_sensorPose);
+
+    /// \brief Visualize the normal forces computed for a (i,j) pixel of the
+    /// image returned by the depth camera
+    /// \param[in] _i First coordinate of the pixel
+    /// \param[in] _j Second coordinate of the pixel
+    /// \param[in] _position Position of the normal force referenced to the
+    /// depth camera's origin
+    /// \param[in] _normalForce Value of the normal force referenced to
+    /// the depth camera's origin
+    /// \param[in] _sensorWorldPose Pose of the plugin's model referenced to
+    /// the world's origin
+    public: void VisualizeNormalForce(uint64_t &_i, uint64_t &_j,
+      ignition::math::Vector3f &_position,
+      ignition::math::Vector3f &_normalForce,
+      ignition::math::Pose3f &_sensorWorldPose);
+
+    /// \brief Message for visualizing the sensor
+    private: ignition::msgs::Marker sensorMarkerMsg;
```

Comment:
> It can introduce error, in that if a marker message is configured a certain way, and is reused for another marker, some of the information might not be overwritten and carry to the second marker.

Hmm, I see your point. However, we've seen the plugin's performance is not that good when it comes to visualization, so continuously creating and deleting will make things worse (significantly?). Besides, I can't think why somebody would overwrite the messages variables to use them for something else. These variables are documented and the structure of the `OpticalTactilePluginVisualization` class makes sense to me (I'll admit I'm probably biased there!)

Reply:
We will keep this comment on hold, until we figure out whether the markers can be published in a list. It is a big hit on performance, and it is not an option - it's a must.

Reusing the variable is not a deliberate act, it is usually an unintended bug, which can happen in future maintenance and additions. Usually, variables that don't need to be class scope are declared at a local scope just as good practice.

Reply:
Following up on https://github.com/ignitionrobotics/ign-gazebo/pull/229#discussion_r468278088 , if that improves the performance a lot, we can demote this to a local variable.

Reply:
> Reusing the variable is not a deliberate act, it is usually an unintended bug, which can happen in future maintenance and additions. Usually, variables that don't need to be class scope are declared at a local scope just as good practice.

I understand. As seen in https://github.com/ignitionrobotics/ign-gazebo/pull/229#discussion_r468634575, publishing a list improves the performance, so I'm demoting to a local variable. Should I do this for the normal forces _and_ the sensor marker? They are published each time the depth camera publishes an image and in each `PostUpdate` call, respectively.

Reply:
I was thinking definitely `positionMarkerMsg` and `forceMarkerMsg`, which are the arrays I believe. The sensor can be kept as a box. I think it is a good idea to keep it local as well.

Reply:
e216bb2